### PR TITLE
Add pre-commit lint wrapper

### PR DESCRIPTION
**Pull request by wyli**
## Summary
- Make `./holohub lint` a thin wrapper around `pre-commit`, including automatic pre-commit installation when missing.
- Move lint tool installation to pre-commit-managed environments and add `cmakelint` as a local hook.
- Preserve manual developer workflows: `--install-dependencies` prefetches hooks but does not install git commit hooks.
- Remove the redundant `check_lint.yml` workflow now that pre-commit.ci owns PR lint status, and replace the README badge.

## Test plan
- `python3 -m pytest utilities/cli/tests/test_cli.py -k "lint" -v`
- `pre-commit run --files .gitignore .pre-commit-config.yaml CONTRIBUTING.md README.md utilities/cli/cli_dev_guide.md utilities/cli/cli_reference.md utilities/cli/holohub.py utilities/cli/tests/CMakeLists.txt utilities/cli/tests/test_cli.py utilities/requirements.lint.txt`
- `./holohub lint --install-dependencies --dryrun`
- `./holohub lint --dryrun`
- `./holohub lint utilities/cli --dryrun`

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Linting unified under pre-commit; CMake linting added.

* **Documentation**
  * CONTRIBUTING and CLI docs updated with pre-commit-driven lint workflow and usage; README CI badge switched to pre-commit.ci.

* **Bug Fixes**
  * Lint now exits gracefully when no config; clearer message when lint cache is unwritable.

* **Chores**
  * Removed prior CI lint workflow; consolidated lint deps under pre-commit; added .npm to .gitignore.

* **Tests**
  * CLI lint tests rewritten to validate pre-commit behavior and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/nvidia-holoscan/holohub/pull/1532)

### CODE REVIEW

File: utilities/cli/holohub.py [hunk 1]
```diff
@@ -1632,220 +1631,131 @@ def handle_autocompletion_list(self, args: argparse.Namespace) -> None:
...
+                    f'  sudo chown -R "$(id -u):$(id -g)" {cache_dir}\n'
+                    f"  sudo rm -rf {cache_dir}"
+                ),
+                file=sys.stderr,
+            )
+            sys.exit(1)
 
-    def _install_lint_deps(self, dry_run: bool = False) -> None:
-        """Install linting dependencies"""
+    def _install_lint_deps(self, dry_run: bool = False, env: Optional[dict] = None) -> None:
```
[the remark this answers is not in the document]
Comment (wyli) [reply utilities/cli/holohub.py]: Addressed in 02e8b76 by updating `_install_lint_deps` to describe installing pre-commit and prefetching hook environments only.

File: utilities/cli/holohub.py [hunk 2]
```diff
@@ -1632,220 +1631,131 @@ def handle_autocompletion_list(self, args: argparse.Namespace) -> None:
...
-                    ):
-                        cmake_lint_failed = True
-
-                if cmake_lint_failed:
-                    exit_code = 1
-
-        if exit_code == 0 and not args.dryrun:
+            files = self._collect_lint_files(target)
+            if not files:
+                print(Color.yellow(f"No files found under {target}; nothing to lint."))
```
[the remark this answers is not in the document]
Comment (wyli) [reply utilities/cli/holohub.py]: Addressed in 02e8b76. Relative lint paths now resolve against `HOLOHUB_ROOT`, and invalid or out-of-repo paths fail fast with a clear error.

File: utilities/cli/holohub.py [hunk 3]
```diff
@@ -1632,220 +1631,131 @@ def handle_autocompletion_list(self, args: argparse.Namespace) -> None:
...
+                    f"  sudo rm -rf {cache_dir}"
+                ),
+                file=sys.stderr,
+            )
+            sys.exit(1)
 
-    def _install_lint_deps(self, dry_run: bool = False) -> None:
-        """Install linting dependencies"""
+    def _install_lint_deps(self, dry_run: bool = False, env: Optional[dict] = None) -> None:
+        """Install pre-commit and register the git hook."""
```
[the remark this answers is not in the document]
Comment (wyli) [reply utilities/cli/holohub.py]: Addressed in 02e8b76 by updating the docstring and keeping the behavior as hook prefetch only (no git hook install).

File: utilities/cli/holohub.py [hunk 4]
```diff
@@ -1856,7 +1766,10 @@ def _install_lint_deps(self, dry_run: bool = False) -> None:
                 str(HoloHubCLI.HOLOHUB_ROOT / "utilities/requirements.lint.txt"),
             ],
             dry_run=dry_run,
+            env=env,
```
[the remark this answers is not in the document]
Comment (wyli) [reply utilities/cli/holohub.py]: Addressed in a7e9708. `_install_lint_deps` now adds `--user` when not running in a virtual environment so the generated script lands in the user install location that the wrapper adds to PATH.

File: utilities/cli/cli_reference.md [hunk 5]
```diff
@@ -442,7 +459,7 @@ Show a compact overview of the development environment: platform, git state, con
 
 - **Platform** — architecture, GPU type/name, CUDA version, Holoscan SDK version
 - **Git** — current branch, short commit hash, dirty/modified file count
-- **Images** — HoloHub-related Docker images with creation time and running status
+- **Containers** — HoloHub-related Docker images with creation time and running status
```
[the remark this answers is not in the document]
Comment (wyli) [reply utilities/cli/cli_reference.md]: Addressed in 02e8b76 by reverting the CLI reference status section back to `Images`, matching the current status command output.

**Review by tbirdso — APPROVED** [review]
LGTM after high-level review. Suggest waiting for others to have time to review as well

**Comment by wyli** [comment]
> LGTM after high-level review. Suggest waiting for others to have time to review as well

thanks, merging this so that it's ready for CLI consolidation
